src/services/GenerateVerifiableCredential.ts

```diff
@@ -0,0 +1,457 @@
+/* eslint-disable no-console */
+import { Logger } from "@aws-lambda-powertools/logger";
+import { AppError } from "../utils/AppError";
+import { HttpCodesEnum } from "../utils/HttpCodesEnum";
+import { YOTI_CHECKS, YotiSessionDocument } from "../utils/YotiPayloadEnums";
+import {
+	VerifiedCredentialEvidence,
+	VerifiedCredentialSubject,
+} from "../utils/IVeriCredential";
+
+export class GenerateVerifiableCredential {
+  readonly logger: Logger;
+
+  private static instance: GenerateVerifiableCredential;
+
+  constructor(logger: Logger) {
+  	this.logger = logger;
+  }
+
+  static getInstance(logger: Logger): GenerateVerifiableCredential {
+  	if (!GenerateVerifiableCredential.instance) {
+  		GenerateVerifiableCredential.instance = new GenerateVerifiableCredential(
+  			logger,
+  		);
+  	}
+  	return GenerateVerifiableCredential.instance;
+  }
+
+	/**
+		Checks if Document contains a chip and if that chip is valid but checking the chip_csca_trusted field
+	*/
+  private doesDocumentContainValidChip(documentType: string, documentAuthenticityCheckBreakdown: { sub_check: string; result: string; }[]): boolean {
+  	const validDocumentTypes = ["PASSPORT", "NATIONAL_ID", "RESIDENCE_PERMIT"];
+	
+  	if (validDocumentTypes.includes(documentType)) {
+  		const validChip = documentAuthenticityCheckBreakdown.some(
+  			(subCheck: { sub_check: string; result: string }) =>
+  				subCheck.sub_check === YotiSessionDocument.CHIP_CSCA_TRUSTED && subCheck.result === YotiSessionDocument.SUBCHECK_PASS,
+  		);
+	
+  		return validChip;
+  	}
+	
+  	return false;
+  }
+	
+
+	/**
+	 The following documents will be scored a 3:
+	 	Non chipped Passports or passports with chips which are un-opened
+		UK or EU Drivers Licence
+		EEA National Identity Card (Without Chip being read)
+		
+	 The following documents will be scored a 4:
+	 	UK Passports or Passports where the chip has been opened
+		BRP
+		EEA National Identity Card (With Chip which has been read)
+	*/
```

Comment:
let's add an explicit link to the JIRA or Confluence page where this requirement is defined?